pkg/registry/scripts/kube-client.js

```diff
@@ -0,0 +1,1033 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function() {
+    "use strict";
+
+    /*
+     * Some notes on the create fields.
+     *
+     * Namespaces should be created first, as they must exist before objects in
+     * them are created.
+     *
+     * Services should be created before pods (or replication controllers that
+     * make pods. This is because of the environment variables that pods get
+     * when they want to access a service.
+     *
+     * Create pods before replication controllers ... corner case, but keeps
+     * things sane.
+     */
+
+    var KUBE = "/api/v1";
+    var OPENSHIFT = "/oapi/v1";
+    var DEFAULT = { api: KUBE, create: 0 };
+    var SCHEMA = flat_schema([
```

Comment:
Sure. But what would we do with such info? It would seem that the caller would need to special case these anyway and load entries completely differently ... using ```loader.load()``` instead of ```loader.watch()```

Can you describe an example?